sdk/provisioning/Azure.Provisioning/src/BicepValue.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Primitives;
+
+namespace Azure.Provisioning;
+
+// TODO: Currently everything's wrapped in BicepValue, but we're going to pull
+// models and enums out (kind of like we do with ResourceReference already) so
+// we get a better intellisense experience.  That'll also remove a lot of the
+// clutter in this class hierarchy.
+
+/// <summary>
+/// Represents the value of a property that could be a literal .NET value, a
+/// Bicep expression, or it could be unset.
+/// </summary>
+public abstract class BicepValue
```

Comment:
`BicepValue` is the root of our primitive types.  It's an abstraction for .NET values, Bicep expressions, or `Unset` which gets interpreted as "just leave this as a symbolic reference."